web_decimal_numpad_dot/static/src/js/numpad_dot.js

```diff
@@ -15,28 +15,29 @@ odoo.define("web_decimal_numpad_dot.FieldFloat", function (require) {
         },
 
         l10n_decimal_point: function () {
-            return this.widget == "float_time"
+            return this.formatType === "float_time"
                 ? ":" : translation._t.database.parameters.decimal_point;
         },
 
         numpad_dot_replace: function (event) {
-            String.prototype.replaceAt=function(index, replacement) {
-                return this.substr(0, index) + replacement + this.substr(index + replacement.length);
-            }
+            String.prototype.replaceAt = function (index, replacement) {
```

Comment:
:scream: WHAT? Were we really altering the String prototype just for this absurdly small use case?

I can't believe it... @chienandalu, do you feel like putting this in a separate function and calling it, instead? :man_facepalming:

Reply:
I guess so :laughing:

Reply:
take a look now